src/rust/terminusdb-community/src/graphql/schema.rs

```diff
@@ -435,22 +481,56 @@ impl<'a, C: QueryableContextType + 'a> GraphQLType for TerminusType<'a, C> {
     where
         DefaultScalarValue: 'r,
     {
-        let (name, frames) = info;
-        let frame = &frames.frames[name];
+        let name = &info.class;
+        let allframes = &info.allframes;
+        let frame = &allframes.frames[name];
         match frame {
             TypeDefinition::Class(d) => Self::generate_class_type(d, info, registry),
             TypeDefinition::Enum(_) => panic!("no enum expected here"),
         }
     }
 }
 
+fn rewind_rdf_list<'a>(instance: &'a dyn Layer, cons_id: u64) -> Option<u64> {
+    if let Some(rdf_rest) = instance.predicate_id(RDF_REST) {
+        let mut cons = Some(cons_id);
+        while let Some(id) = cons {
```

Comment:
this doesn't have to be a while let with a Some, cause you immediately return in the None case. so it could be a simple loop {}